src/App.tsx

```diff
@@ -1,24 +1,233 @@
+    setTodos(visibleTodos);
+
+    setDeletedTodosId([]);
+  };
+
+  const hendeleCheckboxChange = async (id: number, completed: boolean) => {
+    try {
+      await updateTodo(id, { completed: !completed });
+
+      const checkedTodoList = todos.map(todo => {
+        if (todo.id === id) {
+          return { ...todo, completed: !completed };
+        }
+
+        return todo;
+      });
+
+      setTodos(checkedTodoList);
+    } catch {
+      setError(ErrorType.UpdatedError);
+      setIsError(true);
+    }
+  };
+
+  const completedTodos = useMemo(() => {
+    const finishedTodos = todos.filter(todo => todo.completed);
+
+    return finishedTodos.length;
+  }, [todos]);
+
+  const toggleAll = async () => {
+    const totalTodos = todos.length;
+    const are = todos.filter(todo => todo.completed).length === totalTodos;
+
+    setTodos(todos.map(todo => ({ ...todo, completed: !are })));
+
+    try {
+      const notCompletedTodos = todos.filter(todo => todo.completed === false);
+
+      if (notCompletedTodos.length > 0) {
+        notCompletedTodos.forEach(async todo => {
+          updateTodo(todo.id, { completed: !todo.completed });
+        });
+      } else {
+        todos.forEach(async todo => {
+          updateTodo(todo.id, { completed: !todo.completed });
+        });
+      }
+    } catch {
+      setError(ErrorType.UpdatedError);
+      setIsError(true);
+    }
+  };
+
+  useEffect(() => {
+    loadTodosFromServer();
+  }, []);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+
+        <NewTodoField
+          onAdd={uploadTodosOnServer}
+          toggleAll={toggleAll}
+          isUploadError={isUploadError}
+          setError={setError}
+          setIsError={setIsError}
+          isTodoLoading={isTodoLoading}
+        />
+        <TodoList
+          filtredTodos={filtredTodos}
+          isTodoLoading={isTodoLoading}
+          deletedTodosId={deletedTodosId}
+          tempTodo={tempTodo}
+          activeTodoId={activeTodoId}
+          onDelete={hendleRemoveTodo}
+          hendeleCheckboxChange={hendeleCheckboxChange}
+          editTodo={editTodo}
+          updatedTodoID={updatedTodoID}
+        />
+        <TodoFilters
+          todos={todos}
+          setFiltredTodos={setFiltredTodos}
+          clearCompleted={clearCompleted}
+          completedTodos={completedTodos}
+        />
+      </div>
+
+      {isError && (
+        <ErrorNotification
+          error={error}
+          setIsError={setIsError}
+          isError={isError}
+        />
+      )}
```

Comment:
you can always render the `ErrorNotification` component and add the check inside. (check my comments there).

or just replace the check with: `error !== ErrorType.None`